jcg_evaluation/src/main/scala/FingerprintExtractor.scala

```diff
@@ -170,4 +166,91 @@ object FingerprintExtractor {
         }
         new FileWriter(outputFile, false)
     }
+
+    private def generateJSFingerprints(config: JCGConfig): Unit = {
+        if (config.debug) println("[DEBUG] " + config.language + " " + config.inputDir + " " + config.outputDir)
+        println("Extracting JS fingerprints")
+        val adapters = List(JSCallGraphAdapter)
+
+        // create output directories and execute all adapters
+        val outputDir = config.outputDir
+        val adapterOutputDir = config.outputDir
+        executeAdapters(adapters, adapterOutputDir)
+
+        // parse expected call graph for test case from json files
+        val expectedCGs: Array[ExpectedCG] =
+            FileOperations.listFilesDeep(config.inputDir, ".json")
+              .filter(f => f.getAbsolutePath.contains("js"))
+              .map(f => new ExpectedCG(f))
+              .sorted(Ordering.by((f: ExpectedCG) => f.filePath))
+
+        if (config.debug) {
+            println("[DEBUG] expectedCGs:" + expectedCGs.map(_.filePath).mkString(","))
+            val generatedCGFiles = FileOperations.listFilesDeep(adapterOutputDir, ".json").filter(f => f.getAbsolutePath.contains("js"))
+            println("[DEBUG] generatedCGFiles: " + generatedCGFiles.mkString(","))
+        }
+
+        val outputWriter = new BufferedWriter(getOutputTarget(outputDir))
+
+
+        // compare expected and generated call graphs and write results to file
+        for (adapter <- adapters) {
```

Comment:
It might be easier to find a better, more unified way to do this once we unified the Testadapters!
For now I  have split the Extractors into separate objects derived from a common FingerprintExtractor trait that contains common methods